code/weapon/weapons.cpp

```diff
@@ -4929,7 +4911,7 @@ static void weapon_update_state(weapon* wp)
 			infree_flight = true;
 		}
 		else if (lifetime >= fl2f(wip->free_flight_time) &&
-			(lifetime - Frametime) <= fl2f(wip->free_flight_time) && wp->homing_object != nullptr)
+			(lifetime - Frametime) <= fl2f(wip->free_flight_time) && wp->homing_object != &obj_used_list)
```

Comment:
We'll need a comment about the &obj_used_list to make sure it's clear to others in the future.  I myself would feel very confused if I saw that.

Reply:
It's still weird to me, but its something about linked lists. It is used all over the code though, so it seems strange to single out these spots, should I be commenting all of them?

Reply:
*looks around* Just here, I think.